### PR TITLE
remove 1-second join from threadexception plugin

### LINKED ISSUE #13028: remove 1-second join from threadexception plugin
I think it is a reasonable thing for someone to run some thread unrelated to pytest, like a monitoring thread or similar (there are many which do this), and then run `pytest.main()`. If I understand correctly, this would then cause an unnecessary 1s delay to the pytest run. Best case, they report a bug, worst case they just think pytest is slow and suffer through it.

So IMO, unless we can somehow restrict `join_threads` only to threads started by tests, we shouldn't do it.

_Originally posted by @bluetech in https://github.com/pytest-dev/pytest/pull/13016#discussion_r1870340465_

### PR DESCRIPTION
**Pull request by graingert**
fixes #13028 13028 per review comment https://github.com/pytest-dev/pytest/pull/13016#discussion_r1870340465

### CODE REVIEW

File: changelog/13028.improvement.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Remove (unreleased) 1-second join from threadexception plugin.
```
Comment (nicoddemus) [inline changelog/13028.improvement.rst:1]: Given the original change was not released yet, we can skip this changelog entry.

**Review by nicoddemus — APPROVED** [review]

File: src/_pytest/threadexception.py [hunk 2]
```diff
@@ -96,7 +79,9 @@ def cleanup(
 ) -> None:
     try:
         try:
-            join_threads()
+            # we don't join threads here, so exceptions raised from any
```
Comment (nicoddemus) [inline src/_pytest/threadexception.py:82]:
```suggestion
            # We don't join threads here, so exceptions raised from any
```

**Comment by nicoddemus** [comment]
Thanks @graingert for the follow up!

File: src/_pytest/threadexception.py [hunk 3]
```diff
@@ -96,7 +79,9 @@ def cleanup(
 ) -> None:
     try:
         try:
-            join_threads()
+            # We don't join threads here, so exceptions raised from any
+            # threads still running by the time _threading_atexits joins them
+            # do not get captured.
```
Comment (nicoddemus) [inline src/_pytest/threadexception.py:84]:
```suggestion
            # do not get captured (see #13027).
```

**Review by bluetech — APPROVED** [review]
Thanks!

**Pull request merged**